File: common/domain/src/main/java/nl/tecon/cme/common/domain/typehandler/DateTimeTypeHandler.java
package nl.tecon.cme.common.domain.typehandler;

import org.apache.ibatis.type.BaseTypeHandler;
import org.apache.ibatis.type.JdbcType;
import org.joda.time.DateTime;

import java.sql.CallableStatement;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Timestamp;

/**
 * User: x082062 (Thies Edeling - thies@te-con.nl)
 * Date: 12/20/10 3:45 PM
 */
public class DateTimeTypeHandler extends BaseTypeHandler
{
    @Override
    public void setNonNullParameter(PreparedStatement ps, int i, Object parameter, JdbcType jdbcType) throws SQLException
    {
        DateTime dateTime = (DateTime) parameter;

        ps.setTimestamp(i, new Timestamp(dateTime.toDate().getTime()));
    }

    @Override
    public Object getNullableResult(ResultSet rs, String columnName) throws SQLException
    {
        Timestamp timestamp = rs.getTimestamp(columnName);

        return timestamp != null ? new DateTime(timestamp) : null;
    }

    @Override
    public Object getNullableResult(CallableStatement cs, int columnIndex) throws SQLException
    {
        Timestamp timestamp = cs.getTimestamp(columnIndex);

        return timestamp != null ? new DateTime(timestamp) : null;
    }
}

File: domain/src/main/scala/nl/tecon/ucm/domain/ibatis/DateTimeTypeHandler.scala
package nl.tecon.ucm.domain.ibatis

import org.apache.ibatis.`type`.{JdbcType, BaseTypeHandler}
import org.joda.time.DateTime
import java.sql.{PreparedStatement, ResultSet, CallableStatement, Timestamp}

class DateTimeTypeHandler extends BaseTypeHandler {
  override def setNonNullParameter(ps: PreparedStatement, i: Int, parameter: AnyRef, jdbcType: JdbcType) {
    ps.setTimestamp(i, new Timestamp(parameter.asInstanceOf[DateTime].toDate.getTime))
  }

  override def getNullableResult(rs: ResultSet, columnName: String) = if (rs.getTimestamp(columnName) != null) new DateTime(rs.getTimestamp(columnName)) else null

  override def getNullableResult(resultSet: ResultSet, i: Int) = if (resultSet.getTimestamp(i) != null) new DateTime(resultSet.getTimestamp(i)) else null

  override def getNullableResult(cs: CallableStatement, columnIndex: Int) = if (cs.getTimestamp(columnIndex) != null) new DateTime(cs.getTimestamp(columnIndex)) else null
}

